ledger-service/http-json/src/it/scala/http/HttpServiceIntegrationTest.scala

```diff
@@ -66,6 +79,148 @@ abstract class HttpServiceIntegrationTest
+        initialTplId: domain.TemplateId.OptionalPkg,
+        exerciseBy: domain.TemplateId.OptionalPkg,
+    ) = for {
+      aliceH <- getUniquePartyAndAuthHeaders(uri)("Alice")
+      (alice, aliceHeaders) = aliceH
+      createTest <- postCreateCommand(
+        iouCommand(alice, initialTplId),
+        encoder,
+        uri,
+        aliceHeaders,
+      )
+      testIIouID = {
+        discard { createTest._1 should ===(StatusCodes.OK) }
+        createTest._2.convertTo[domain.OkResponse[domain.ActiveContract[JsValue]]].result.contractId
+      }
+      bobH <- getUniquePartyAndAuthHeaders(uri)("Bob")
+      (bob, _) = bobH
+      exerciseTest <- postJsonRequest(
+        uri withPath Uri.Path("/v1/exercise"),
+        encodeExercise(encoder)(
+          iouTransfer(domain.EnrichedContractId(Some(exerciseBy), testIIouID), bob)
+        ),
+        aliceHeaders,
+      )
+    } yield exerciseTest._1 should ===(StatusCodes.OK)
+
+    for {
+      // _ <- uploadPackage(uri)(iiouDar)
+      _ <- uploadPackage(uri)(ciouDar)
+      // first, use IIou only
+      _ <- createIouAndExerciseTransfer(
+        initialTplId = domain.TemplateId(None, "IIou", "TestIIou"),
+        // whether we can exercise by interface-ID
+        exerciseBy = iiouIfaceID,
+      )
+      // next, use CIou
+      _ <- createIouAndExerciseTransfer(
+        initialTplId = domain.TemplateId(None, "CIou", "CIou"),
+        // whether we can exercise inherited by concrete template ID
+        exerciseBy = domain.TemplateId(None, "CIou", "CIou"),
+      )
+    } yield succeed
+  }
+
+  "fail to exercise by key with interface ID" in withHttpService { (uri, encoder, _, _) =>
+    import json.JsonProtocol._
+    for {
+      _ <- uploadPackage(uri)(ciouDar)
+      aliceH <- getUniquePartyAndAuthHeaders(uri)("Alice")
+      (alice, aliceHeaders) = aliceH
+      createTest <- postCreateCommand(
+        iouCommand(alice, domain.TemplateId(None, "CIou", "CIou")),
+        encoder,
+        uri,
+        aliceHeaders,
+      )
+      _ = createTest._1 should ===(StatusCodes.OK)
+      bobH <- getUniquePartyAndAuthHeaders(uri)("Bob")
+      (bob, _) = bobH
+      exerciseTest <- postJsonRequest(
+        uri withPath Uri.Path("/v1/exercise"),
+        encodeExercise(encoder)(
+          iouTransfer(
+            domain.EnrichedContractKey(
+              iiouIfaceID,
+              v.Value(v.Value.Sum.Party(domain.Party unwrap alice)),
+            ),
+            bob,
+          )
+        ),
+        aliceHeaders,
+      )
+    } yield {
+      val Status = StatusCodes.BadRequest
+      discard { exerciseTest._1 should ===(Status) }
+      inside(exerciseTest._2.convertTo[domain.ErrorResponse]) {
+        case domain.ErrorResponse(Seq(lookup), None, Status) =>
+          lookup should include regex raw"Cannot resolve Template Key type, given: TemplateId\([0-9a-f]{64},IIou,IIou\)"
+      }
+    }
+  }
+
+  // TODO(SC #13301) test against DB too
```

Comment:
Ideally tests that aren't affected by DB ought to be here; the original split in the `HttpServiceIntegrationTest` was to run some tests against DB. The corollary being that _this_ test in particular might be treated differently on a DB deployment so should be abstracted instead.